src/models/StyledComponent.js

```diff
@@ -203,21 +186,27 @@ export default function createStyledComponent(target: Target, options: Object, r
 
   const styledComponentId =
     options.displayName && options.componentId
-      ? `${escape(options.displayName)}-${options.componentId}`
+      ? `${escape(options.displayName)}-${String(options.componentId)}`
       : options.componentId || componentId;
 
   // fold the underlying StyledComponent attrs up (implicit extend)
   const finalAttrs =
     // $FlowFixMe
-    isTargetStyledComp && target.attrs ? { ...target.attrs, ...attrs } : attrs;
+    isTargetStyledComp && target.attrs
+      ? (context: Context): Object => ({
+          // $FlowFixMe <- target.attrs
+          ...processAttrs(target.attrs, context),
+          ...processAttrs(attrs, context),
```

Comment:
Maybe creating of this function here and in constructor can be extracted as `createAttrsResolver` and use typings.

Reply:
Sure, I will update this PR later.